### PR TITLE
Add Node, Theorem, Proof Model/Classes

### PR DESCRIPTION
**Pull request by Bera0422**
Add Node, Theorem and Proof Model/Classes with unit tests. Dependent models of them are also created with no content.

### CODE REVIEW

File: project/backend/database/models.py [hunk 1]
```diff
@@ -3,11 +3,71 @@
...
+    theorem = models.OneToOneField(Theorem, null=True, on_delete=models.SET_NULL)
+    publish_date = models.DateField()
+    reviewers = models.ManyToManyField(Reviewer)
+    referenced_nodes = models.ManyToManyField("self", symmetrical=True)
+    semantic_tags = models.ManyToManyField(SemanticTag)
+    wiki_tags = models.ManyToManyField(WikiTag)
+    annotations = models.ManyToManyField(Annotation)
+    is_valid = models.BooleanField()
+    num_visits = models.IntegerField()
+
```
Comment (hakanaktas0) [inline project/backend/database/models.py:55]: Proofs should be added as a one-to-many relation.
Nodes that have referenced this node should be added as a many to many relation to 'self'.

**Review by hakanaktas0 — CHANGES_REQUESTED** [review]
This PR can be merged after the necessary additions are made to Node model

File: project/backend/database/models.py (hunk 1 above)
Comment (Bera0422) [reply project/backend/database/models.py:61]: I updated the code accordingly in the next commit.

**Review by hakanaktas0 — APPROVED** [review]
All looks good

**Review by gebenalimert — APPROVED** [review]
All things seems to be working smoothly.

**Pull request merged**